### PR TITLE
Add link to profiling recipe from rpc main docs

### PR DESCRIPTION
**Pull request by rohan-varma**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#45235 Add link to profiling recipe from rpc main docs**

This is so that users know that the profiler works as expected with
RPC and they can learn how to use it to profile RPC-based workloads.

Screenshot:
profiler_ss

Differential Revision: [D23777888](https://our.internmc.facebook.com/intern/diff/D23777888/)

### CODE REVIEW

**Comment by rohan-varma** [comment]
profiler_ss

File: docs/source/rpc.rst [hunk 1]
```diff
@@ -293,8 +293,11 @@ The RRef design note covers the design of the :ref:`rref` (Remote REFerence) pro
 
 Tutorials
 ---------
-The RPC tutorial introduces users to the RPC framework and provides two example applications using :ref:`torch.distributed.rpc<distributed-rpc-framework>` APIs.
+The RPC tutorials introduce users to the RPC framework, provide several example applications
+using :ref:`torch.distributed.rpc<distributed-rpc-framework>` APIs, and demonstrate how
+to use `the profiler <https://pytorch.org/docs/stable/autograd.html>`__ to profile RPC-based workloads.
```
Comment (mrshenli) [inline docs/source/rpc.rst:298]: Will the link below be more accurate?

```
https://pytorch.org/docs/stable/autograd.html#profiler
```

**Review by mrshenli — APPROVED** [review]
Thanks for adding this!

File: docs/source/rpc.rst (hunk 1 above)
Comment (rohan-varma) [reply docs/source/rpc.rst:298]: Yes, thanks, will update!

**Pull request closed without merging**